Mage.Sets/src/mage/cards/g/GaleWaterdeepProdigy.java

```diff
@@ -0,0 +1,196 @@
+package mage.cards.g;
+
+import mage.ApprovingObject;
+import mage.MageInt;
+import mage.abilities.Ability;
+import mage.abilities.common.ChooseABackgroundAbility;
+import mage.abilities.common.SpellCastControllerTriggeredAbility;
+import mage.abilities.effects.ContinuousEffect;
+import mage.abilities.effects.OneShotEffect;
+import mage.abilities.effects.ReplacementEffectImpl;
+import mage.cards.Card;
+import mage.cards.CardImpl;
+import mage.cards.CardSetInfo;
+import mage.constants.*;
+import mage.filter.FilterCard;
+import mage.filter.common.FilterInstantOrSorcerySpell;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.events.ZoneChangeEvent;
+import mage.game.stack.Spell;
+import mage.players.Player;
+import mage.target.common.TargetCardInYourGraveyard;
+import mage.target.targetpointer.FixedTarget;
+import mage.watchers.common.CastFromHandWatcher;
+
+import java.util.UUID;
+
+import static mage.cards.g.GaleWaterdeepProdigy.INSTANT_FILTER;
+import static mage.cards.g.GaleWaterdeepProdigy.SORCERY_FILTER;
+
+/**
+ * @author Rjayz
+ */
+public final class GaleWaterdeepProdigy extends CardImpl {
+
+    static final FilterCard SORCERY_FILTER = new FilterCard();
+    static final FilterCard INSTANT_FILTER = new FilterCard();
```

Comment:
you can put these directly into `GaleWaterdeepProdigyTriggeredAbility` as private, also you can put the message directly into the constructor instead of calling `setMessage`